src/Xamarin.Android.Build.Tasks/Utilities/AssemblyCompression.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Buffers;
+using System.IO;
+
+using K4os.Compression.LZ4;
+
+namespace Xamarin.Android.Tasks
+{
+	class AssemblyCompression
+	{
+		public sealed class AssemblyData
+		{
+			public string SourcePath { get; internal set; }
+			public uint DescriptorIndex { get; internal set; }
+
+			public string DestinationPath;
+			public uint SourceSize;
+			public uint DestinationSize;
+
+			public AssemblyData (string sourcePath, uint descriptorIndex)
+			{
+				SetData (sourcePath, descriptorIndex);
+			}
+
+			public void SetData (string sourcePath, uint descriptorIndex)
+			{
+				if (String.IsNullOrEmpty (sourcePath))
+					throw new ArgumentException ("must not be null or empty", nameof (sourcePath));
+				SourcePath = sourcePath;
+				DescriptorIndex = descriptorIndex;
+			}
+		}
+
+		const uint CompressedDataMagic = 0x5A4C4158; // 'XALZ', little-endian
+
+		static readonly ArrayPool<byte> bytePool = ArrayPool<byte>.Create ();
```

Comment:
I think you should just use `ArrayPool<byte>.Shared`, so it will supposedly share `byte[]`'s with the common pool the BCL uses.

https://github.com/xamarin/LibZipSharp/blob/05c43d01f93a21e56294716faffca49c1deb0450/ZipArchive.cs#L750-L757